src/dispatching/dialogue/storage/sqlite_storage.rs

```diff
@@ -63,35 +63,27 @@ where
     fn remove_dialogue(
         self: Arc<Self>,
         chat_id: i64,
-    ) -> BoxFuture<'static, Result<Option<D>, Self::Error>> {
+    ) -> BoxFuture<'static, Result<(), Self::Error>> {
         Box::pin(async move {
-            Ok(match get_dialogue(&self.pool, chat_id).await? {
-                Some(d) => {
-                    let prev_dialogue =
-                        self.serializer.deserialize(&d).map_err(SqliteStorageError::SerdeError)?;
-                    sqlx::query("DELETE FROM teloxide_dialogues WHERE chat_id = ?")
-                        .bind(chat_id)
-                        .execute(&self.pool)
-                        .await?;
-                    Some(prev_dialogue)
-                }
-                _ => None,
-            })
+            if get_dialogue(&self.pool, chat_id).await?.is_some() {
```

Comment:
Unrelated to the PR, but this check seems weird, isn't it better to do it in SQL itself?..

Reply:
Yes, I'll try to refactor it in SQL later, in a separate PR.